### ISSUE TITLE
Rollups not compatible with CRMFusion Dupeblocker

### ISSUE THREAD

**Issue by dboelig** [afawcett/declarative-lookup-rollup-summaries#114]
Hi Andrew - First off, love this tool! It has allowed me to create so many key reports that I was previously unable to do. My organization uses CRMFusion Dupeblocker and since I rolled out the Rollup functionality, the Dupeblocker blocking rules have stopped being enforced. I got in touch with CRMFusion and they informed me that the triggers from the Rollup application were causing the Dupeblocker rules to fail...they sent me to this document: http://na19.salesforce.com/_ui/selfservice/pkb/PublicKnowledgeSolution/d?orgId=00D300000000RBY&id=50130000000Ukao&retURL=%2Fsol%2Fpublic%2Fsolutionbrowser.jsp%3Fsearch%3Dcustom%2Btriggers%26cid%3D000000000000000%26orgId%3D00D300000000RBY%26t%3D4&ps=1 

Now, I am not a developer, but am wondering if there is any way to work around this that you know of? 

Thank you again! 

-Derek

**Comment by afawcett** [comment]
I've taken a look at this, their advice is most odd, they are basically saying Apex triggers cannot immediately insert or update "other objects" in the same context as the Dupeblocker. And have to push this work off into the background (which in doing so runs the risk of errors simply being swallowed or lost for the end user if error handling is not done correctly, which is not shown in the sample). 

Basically asking developers who have written perfectly good Apex trigger code to modify their code to switch to background process is quite an ask for what appears to be no reasoning at all as far as i can see in this doc? I'd like to know a little bit more why they are giving this advice tbh.  

In the meantime, i am curious, what error are you seeing surfaced from Dupeblocker?

**Comment by afawcett** [comment]
Please do let me know if you get any further details on this, keen to help! :+1:

**Comment by dboelig** [comment]
Hey Andrew! I appreciate you getting back to me. This app has opened the
door to tons of analysis we can do in tool now instead of in excel.

I am working with CRMFusion to understand what exactly the cause of the
error is and will definitely get you the details when I have more
information. Will be in touch this week I hope!

Thanks,
Derek

Derek Boelig
CRM Manager
Grovo Learning, Inc.

Office: 1-929-270-0080
Mobile: 603-361-0317
Email: dboelig@grovo.com
www.grovo.com

**Comment by afawcett** [comment]
Awesome! :+1:

**Comment by dboelig** [comment]
Hi Andrew,

Apologies for the long delay getting back. We have actually turned off Dupe
Blocker and will look into alternative ways to manage duplicates (including
SFDCs new functionality.) Thanks again for your response! Much appreciated
and we continue to use your app every day.

Derek Boelig
CRM Manager
Grovo Learning, Inc.

Office: 1-929-270-0080
Mobile: 603-361-0317
Email: dboelig@grovo.com
www.grovo.com

**Comment by afawcett** [comment]
Thanks @dboelig your very welcome, thanks for coming back to me.
